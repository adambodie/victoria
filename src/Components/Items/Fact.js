import React from 'react';

const Fact = props => (
	<div className="photo slide">
		<img src={'https://s3-us-west-2.amazonaws.com/victoria.bodiewebdesign.com/Images/fact.jpg'} alt="Did You Know?"/>
			<div className="photo-overlay">
				<h3>{props.name}</h3> 
			</div>
		</div>
);

export default Fact;
