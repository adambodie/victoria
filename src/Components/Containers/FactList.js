import React from 'react';
import Fact from '../items/Fact';

const FactList = ({ data }) => (
	<div className="facts">
		{data.map((x, index) => 
			<div key={index}>
				<Fact name={x}/>
			</div>
			).filter((x, index) => index < 4)
		}
	</div>
);


export default FactList;
