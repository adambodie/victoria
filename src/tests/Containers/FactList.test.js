import React from 'react';
import { shallow, mount, render } from 'enzyme';
import FactList from '../../Components/Containers/FactList';
import Fact from '../../Components/Items/Fact';


describe('FactList', () => {
  const facts = ["Hall of Fame Football player John Elway was born in Port Angeles", 
				"Congregation Emanu-El is the oldest Jewish Synagogue in Canada",
				"The population of Port Angeles is about 19,500"];
  it('renders 3 facts', () => {
    const wrapper = shallow(<FactList data={facts} />);
    expect(wrapper.find(Fact).length).toBe(facts.length);
    //console.log(wrapper.debug());
  });
});
