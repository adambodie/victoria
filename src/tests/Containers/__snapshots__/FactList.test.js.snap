// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactList renders a fact 1`] = `
Array [
  <div
    className="factButtons"
>
    <button
        onClick={[Function]}
    >
        Less
    </button>
    <button
        onClick={[Function]}
    >
        More
    </button>
</div>,
  <div
    className="facts"
>
    <div
        className="view view-transform"
    >
        <img
            alt="Did You Know?"
            src="..//assets/fact.jpg"
        />
        <div
            className="mask"
        >
            <h2>
                Hall of Fame Football player John Elway was born in Port Angeles
            </h2>
        </div>
    </div>
    <div
        className="view view-transform"
    >
        <img
            alt="Did You Know?"
            src="..//assets/fact.jpg"
        />
        <div
            className="mask"
        >
            <h2>
                Congregation Emanu-El is the oldest Jewish Synagogue in Canada
            </h2>
        </div>
    </div>
    <div
        className="view view-transform"
    >
        <img
            alt="Did You Know?"
            src="..//assets/fact.jpg"
        />
        <div
            className="mask"
        >
            <h2>
                The population of Port Angeles is about 19,500
            </h2>
        </div>
    </div>
</div>,
]
`;
